CodeGeneration/Sources/generate-swiftideutils/SyntaxClassificationFile.swift

```diff
@@ -0,0 +1,116 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+// Collects the list of classifications to use for contextual classification.
+var node_child_classifications: [ChildClassification] {
+  var result = [ChildClassification]()
+  for node in SYNTAX_NODES {
+    for (index, child) in node.children.enumerated() where child.classification?.name != nil {
+      result.append(
+        ChildClassification(node: node, childIndex: index, child: child)
+      )
+    }
+  }
+  
+  return result
+}
+
+let syntaxClassificationFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    attributes: AttributeList {
+      AttributeSyntax(
+        attributeName: TokenSyntax.identifier("_spi"),
+        leftParen: .leftParen,
+        argument: .identifier(.identifier("RawSyntax")),
+        rightParen: .rightParen)
+    },
+    path: [AccessPathComponent(name: "SwiftSyntax")]
+  )
+  
+  EnumDeclSyntax("public enum SyntaxClassification") {
+    for classification in SYNTAX_CLASSIFICATIONS {
+      EnumCaseDecl("/// \(classification.description)\ncase \(classification.swiftName)")
```

Comment:
I think I would prefer a multi-line string literal here. Makes it easier to read IMO.